src/core/definition_directory.ts

```diff
@@ -0,0 +1,223 @@
+    this.path = resolve(directory);
+    this.definitions = [];
+    // Transform basic patterns '*' or '{text}' into a real RegExp
+    this.filenamePattern = new RegExp(
+      '^' + filenamePattern.replace('*', '.+?').replace(/{.*?}/, '(?<slug>.+?)') + '$',
+    );
+
+    this.buildNewFilename = (slug) =>
+      filenamePattern.replace('*', '').replace(/{.*?}/, slug);
+
+    this.humanFilenamePattern = filenamePattern.replace(
+      /{(.*?)}/,
+      `${chalk.inverse('{$1}')}`,
+    );
+  }
+
+  public async readDefinitions(): Promise<DefinitionConfig[]> {
+    for await (const { value, filename } of File.listValidConventionFiles(
+      this.path,
+      this.filenamePattern,
+    )) {
+      const file = join(this.path, value);
+      /* We already check the filenamePattern match inside the
+      `File.listValidConventionFiles` method so we are sure the group
+      matched exists. */
+      /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+      const slug = filename.match(this.filenamePattern)!.groups!.slug!;
+      const definition = await API.load(file);
+      this.definitions.push({
+        file,
+        definition,
+        slug,
+      });
+    }
+
+    if (this.definitions.length) {
+      cli.info(
+        chalk.underline(
+          `We've found ${this.definitions.length} valid API definitions to deploy`,
+        ),
+      );
+      const subtree = cli.tree();
+      this.definitions.forEach(({ file, definition }) =>
+        subtree.insert(
+          `${basename(file)} (${definition.specName} spec version ${definition.version})`,
+        ),
+      );
+      const tree = cli.tree();
+      tree.insert(this.path, subtree);
+
+      tree.display();
+      cli.info('');
+    }
+
+    return this.definitions;
+  }
+
+  public definitionsExists(): boolean {
+    return !!this.definitions.length;
+  }
+
+  public async sequentialMap(
+    callback: (definition: DefinitionConfig) => Promise<DefinitionConfig>,
+  ): Promise<void> {
+    for (const definition of this.definitions) {
+      await callback(definition);
+    }
+
+    return;
+  }
+
+  public async renameToConvention(documentation: DefinitionConfig): Promise<void> {
+    const { file, slug } = documentation;
+    // Default convention is defined in the flags.ts file for the
+    // 'filenamePattern' flag.
+    const newFilename = this.buildNewFilename(slug);
+    const newFile = `${dirname(file)}/${newFilename}${extname(file)}`;
+
+    if (file === newFile) return;
+
+    let confirm = true;
+    await this.confirm(
+      `Do you want to rename ${file} to ${newFile} (for later deployments)?`,
+    ).catch(() => {
+      confirm = false;
+    });
+
+    if (confirm) {
+      await rename(file, newFile, (err) => {
+        if (err) throw err;
+        cli.styledSuccess(`Renamed ${file} to ${newFile}.`);
+      });
+    }
+
+    return;
+  }
+
+  public async confirm(message = 'Continue?'): Promise<void> {
```

Comment:
I would also move this is a file dedicated prompt functionalities.

Reply:
I suggested that the "public async confirm" method may be moved in a file : ./utils/prompt.ts because it's not really an action dedicated to a "definition directory"